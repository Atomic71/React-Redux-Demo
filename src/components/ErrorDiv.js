import React from 'react';
import {connect} from 'react-redux';
import {CSSTransition} from 'react-transition-group';
import PropTypes from 'prop-types';

import {hideErrorDiv} from '../store/actions/error/index';

const MapStateToProps = (state) => ({ error: state.errors.div });
const MapDispatchToProps = (dispatch) => ({ hideError: () => dispatch(hideErrorDiv()) })


export const ErrorDiv = ({ error, hideError }) => (
    <CSSTransition
        mountOnEnter
        unmountOnExit
        timeout={300} 
        in={error ? true : false} 
        classNames="slide" >
        <div key={error} className="error-div">
            <p className="error-text">
                {error}
            </p>
            <span onClick={hideError} className="error-close">&#10005;</span>
        </div>
    </CSSTransition>
)

ErrorDiv.propTypes = {
    error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]).isRequired,
    hideError: PropTypes.func.isRequired
}


export const ErrorDivConnected = connect(MapStateToProps, MapDispatchToProps)(ErrorDiv)