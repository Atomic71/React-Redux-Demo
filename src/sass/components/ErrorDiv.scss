.ErrorDiv {
        display: flex;
        justify-content: space-between;
        align-self: stretch;
        padding: .5rem 1.5rem;
        background: orangered;
        color: $white;
        font-size: 1.2rem;
        align-items: center;
        position: absolute;
        top: 0;
        left: 0;
        right: 0;
        z-index: 200;
    &-text {
        font-size: 1.2rem;
    }
    &-close {
        cursor: pointer;
        transition: all .1s;
        &:hover {
            color: $dark;
        }
    }
}

.slide-enter {
    opacity: 0;
    transform: translateY(-100%)
}

.slide-enter-active {
    transform: translateY(0);
    opacity: 1;
    transition: all .3s;
}

.slide-exit-active {
    opacity: 0;
    transition: all .3s;
}