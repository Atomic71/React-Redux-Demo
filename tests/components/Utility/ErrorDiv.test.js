import React from 'react';
import { ErrorDiv } from "../../../src/components/Utility/ErrorDiv";
import { CSSTransition } from 'react-transition-group';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


describe('<ErrorDiv />', () => {
    const myMockFn = jest.fn();
    it('does not render when error is false' , () => {
        const tree = renderer.create(
            <ErrorDiv error={false} hideError={myMockFn} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
    it('does render when there is an error', ()=> {
        const tree = renderer.create(
            <ErrorDiv error="Some error" hideError={myMockFn} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('calls the onCLick properly', () => {
        const myErrDiv = shallow(<ErrorDiv error="Some error" hideError={myMockFn}/>)
        const closeBtn = myErrDiv.find('span');
        closeBtn.simulate('click');
        expect(myMockFn.mock.calls.length).toBe(1);        
    })

})