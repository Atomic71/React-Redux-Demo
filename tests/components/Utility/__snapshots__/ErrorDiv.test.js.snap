// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorDiv /> does not render when error is false 1`] = `null`;

exports[`<ErrorDiv /> does render when there is an error 1`] = `
<div
  className="error-div"
>
  <p
    className="error-text"
  >
    Some error
  </p>
  <span
    className="error-close"
    onClick={[MockFunction]}
  >
    âœ•
  </span>
</div>
`;
